Building the hsm Docker image often times out

Hi,

I first started following the steps here:
https://smallstep.com/blog/build-a-tiny-ca-with-raspberry-pi-yubikey/

I tried following along with the v0.22.1 tag from GitHub with my Raspberry Pi OS (September 2022 release), but the `make build` instruction didn't seem to do anything. I learned later that some go modules depend on golang higher than v1.18, but this didn't show in the console at the time.

I eventually went looking for the `docker/Dockerfile.step-ca.hsm` file, which seems to yield essentially the same result (for me at least). What I was missing following the tutorial was the `-x` flag for `go mod download`. The order of parameters is also slightly different from the blog post: `make V=1 GOFLAGS="" build`.

Anyway, the purpose of this issue was to draw your attention to the way the Docker image is built: a whole lot of downloads are processed in one operation (`go mod download`) within another larger operation (`make build`). This isn't cacheable by `docker build`, it might pop different issues and all of this happens with unreliable error reporting (the `-x` flag). It might work on a CI server, but on-device is a different matter (ie. the blog post).

In my case, I was trying to get tiny-ca running on a Raspberry Pi (as described in the blog post) with flash storage, so writes are really slow. Timeouts were going to happen, and indeed, the download operation has failed on me twice.

```
$ go version
go version go1.15.15 linux/arm64
```

```
go: google.golang.org/grpc@v1.47.0: Get "https://proxy.golang.org/google.golang.org/grpc/@v/v1.47.0.zip": net/http: TLS handshake timeout
make: *** [Makefile:92: download] Error 1
The command '/bin/sh -c make V=1 GOFLAGS="" build' returned a non-zero code: 2
```

I eventually got an image tagged:
```
Successfully built 14afbe3cb557
Successfully tagged stepca:tmp01
```

I am not sure what to suggest, perhaps splitting the Dockerfile operations?
```
RUN apk add --no-cache curl git make
RUN apk add --no-cache gcc musl-dev pkgconf pcsc-lite-dev
RUN make V=1 GOFLAGS="" download
RUN make V=1 GOFLAGS="" build
```

--- a/docker/Dockerfile.step-ca
+++ b/docker/Dockerfile.step-ca
@@ -4,6 +4,7 @@ WORKDIR /src
 COPY . .
 
 RUN apk add --no-cache curl git make
+RUN make V=1 download
 RUN make V=1 bin/step-ca bin/step-awskms-init bin/step-cloudkms-init
 
 

--- a/docker/Dockerfile.step-ca.hsm
+++ b/docker/Dockerfile.step-ca.hsm
@@ -5,6 +5,7 @@ COPY . .
 
 RUN apk add --no-cache curl git make
 RUN apk add --no-cache gcc musl-dev pkgconf pcsc-lite-dev
+RUN make V=1 download
 RUN make V=1 GOFLAGS="" build
 
 

